pkg/lineprinter/lineprinter.go

```diff
@@ -0,0 +1,70 @@
+// Package lineprinter wraps a Print implementation to provide an io.WriteCloser.
+package lineprinter
+
+import (
+	"bytes"
+	"io"
+	"sync"
+)
+
+// Print is a type that can hold fmt.Print and other implementations
+// which match that signature.  For example, you can use:
+//
+//   linePrinter := &linePrinter{Print: logrus.StandardLogger().Debug}
+//
+// to connect the line printer to logrus at the debug level.
+type Print func(args ...interface{})
+
+// LinePrinter is an io.WriteCloser that buffers written bytes.
+// During each Write, newline-terminated lines are removed from the
+// buffer and passed to Print.  On Close, any content remaining in the
+// buffer is also passed to Print.
+//
+// One use-case is connecting a subprocess's standard streams to a
+// logger:
+//
+//   linePrinter := &linePrinter{Print: logrus.StandardLogger().Debug}
+//   defer linePrinter.Close()
+//   cmd := exec.Command(...)
+//   cmd.Stdout = linePrinter
+type LinePrinter struct {
+	buf   bytes.Buffer
+	lock  sync.Mutex
+	Print Print
+}
+
+// Write writes len(p) bytes from p to an internal buffer.  Then it
+// retrieves any newline-terminated lines from the internal buffer and
+// prints them with lp.Print.  Partial lines are left in the internal
```

Comment:
Do you think it is useful to document why partial line are left in internal buffer? 
Or its common thing that it allows wait for newlines in next write until close?

Reply:
> Do you think it is useful to document why partial line are left in internal buffer?

What sort of wording were you looking for?  I expect Go's `Cmd` is treating the subprocess streams as opaque byte streams.  Logging, on the other hand, is line-based.  So we need logic to buffer the byte stream and split it into lines for the logger.  Sometimes we might have a partial line written to the buffer (I haven't looked into the code, but I expect there's a goroutine that collects up to `$SIZE` bytes in a select switch or some such).  We don't want to push that partial line into the logs if the next read attempt will pull in the remainder of the line.  But we do want to push that partial line into the logs if there will never be a next read.  Did you have a compact way to express all of that?

Reply:
doesn't have to be compact. :P Add that ^^